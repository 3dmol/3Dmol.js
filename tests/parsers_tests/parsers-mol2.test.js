
global.$ = require("jquery");
global.URL.createObjectURL = function() {};
let $3Dmol = require("../../build/3Dmol.js");
const fs = require('fs');
const path = require('path');

describe('Function MOL2 | Input: multiple.mol2 |', ()=>{
    const data = fs.readFileSync(path.resolve(__dirname,'../test_structs/multiple.mol2'), 'utf-8')
    let atoms = $3Dmol.Parsers.MOL2(data, {});// cannot use multimodel

    test("Atoms length is 1", ()=>{
        expect(atoms.length).toBe(1);
    });
    
    test("Atoms should match the snapshot", ()=>{
        expect(atoms).toMatchSnapshot();
    });
    
});

// noH
describe('Function MOL2 | Input: multiple.mol2 | options: keepH |', ()=>{
    const data = fs.readFileSync(path.resolve(__dirname,'../test_structs/multiple.mol2'), 'utf-8')
    let atoms = $3Dmol.Parsers.MOL2(data, {keepH:false}); // noH = true
    
    test("Atoms should match the snapshot (keepH: false)", ()=>{
        expect(atoms).toMatchSnapshot();
    });
    
});
