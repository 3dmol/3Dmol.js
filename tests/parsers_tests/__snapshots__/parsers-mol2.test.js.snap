// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function MOL2 | Input: multiple.mol2 | Atoms should match the snapshot 1`] = `
[
  [
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        28,
        1,
        5,
      ],
      "elem": "C",
      "index": 0,
      "properties": {
        "charge": 0.2586,
        "partialCharge": 0.2586,
      },
      "serial": 1,
      "x": 5.9063,
      "y": 24.9584,
      "z": 25.9808,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        0,
        6,
        2,
      ],
      "elem": "C",
      "index": 1,
      "properties": {
        "charge": 0.1841,
        "partialCharge": 0.1841,
      },
      "serial": 2,
      "x": 4.7041,
      "y": 24.1561,
      "z": 26.7625,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        1,
        10,
        3,
      ],
      "elem": "C",
      "index": 2,
      "properties": {
        "charge": 0.2062,
        "partialCharge": 0.2062,
      },
      "serial": 3,
      "x": 4.3137,
      "y": 24.9723,
      "z": 28.0348,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        2,
        13,
        4,
      ],
      "elem": "C",
      "index": 3,
      "properties": {
        "charge": 0.1795,
        "partialCharge": 0.1795,
      },
      "serial": 4,
      "x": 5.6843,
      "y": 25.1844,
      "z": 28.7746,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        3,
        5,
        11,
      ],
      "elem": "C",
      "index": 4,
      "properties": {
        "charge": 0.1803,
        "partialCharge": 0.1803,
      },
      "serial": 5,
      "x": 6.7356,
      "y": 26.0149,
      "z": 27.8889,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        0,
        4,
      ],
      "elem": "O",
      "index": 5,
      "properties": {
        "charge": -0.3263,
        "partialCharge": -0.3263,
      },
      "serial": 6,
      "x": 7.1258,
      "y": 25.0806,
      "z": 26.8068,
    },
    {
      "atom": "N.am",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        1,
        7,
      ],
      "elem": "N",
      "index": 6,
      "properties": {
        "charge": -0.1992,
        "partialCharge": -0.1992,
      },
      "serial": 7,
      "x": 3.5113,
      "y": 23.9905,
      "z": 25.9254,
    },
    {
      "atom": "C.2",
      "bondOrder": [
        1,
        2,
        1,
      ],
      "bonds": [
        6,
        8,
        9,
      ],
      "elem": "C",
      "index": 7,
      "properties": {
        "charge": 0.242,
        "partialCharge": 0.242,
      },
      "serial": 8,
      "x": 3.1108,
      "y": 22.8071,
      "z": 25.3967,
    },
    {
      "atom": "O.2",
      "bondOrder": [
        2,
      ],
      "bonds": [
        7,
      ],
      "elem": "O",
      "index": 8,
      "properties": {
        "charge": -0.2733,
        "partialCharge": -0.2733,
      },
      "serial": 9,
      "x": 3.7384,
      "y": 21.8496,
      "z": 25.5073,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        7,
      ],
      "elem": "C",
      "index": 9,
      "properties": {
        "charge": 0.0964,
        "partialCharge": 0.0964,
      },
      "serial": 10,
      "x": 1.8037,
      "y": 22.8588,
      "z": 24.6767,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        2,
      ],
      "elem": "O",
      "index": 10,
      "properties": {
        "charge": -0.2184,
        "partialCharge": -0.2184,
      },
      "serial": 11,
      "x": 3.3693,
      "y": 24.2052,
      "z": 28.8011,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        4,
        12,
      ],
      "elem": "C",
      "index": 11,
      "properties": {
        "charge": 0.2141,
        "partialCharge": 0.2141,
      },
      "serial": 12,
      "x": 7.9504,
      "y": 26.3491,
      "z": 28.7097,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        11,
      ],
      "elem": "O",
      "index": 12,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 13,
      "x": 8.2497,
      "y": 25.1328,
      "z": 29.5559,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        3,
        14,
      ],
      "elem": "O",
      "index": 13,
      "properties": {
        "charge": -0.3264,
        "partialCharge": -0.3264,
      },
      "serial": 14,
      "x": 5.5738,
      "y": 25.9109,
      "z": 30.0347,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        13,
        15,
        19,
      ],
      "elem": "C",
      "index": 14,
      "properties": {
        "charge": 0.2586,
        "partialCharge": 0.2586,
      },
      "serial": 15,
      "x": 4.5765,
      "y": 26.6958,
      "z": 30.7444,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        14,
        20,
        16,
      ],
      "elem": "C",
      "index": 15,
      "properties": {
        "charge": 0.1842,
        "partialCharge": 0.1842,
      },
      "serial": 16,
      "x": 3.837,
      "y": 27.7677,
      "z": 29.9491,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        15,
        25,
        17,
      ],
      "elem": "C",
      "index": 16,
      "properties": {
        "charge": 0.2094,
        "partialCharge": 0.2094,
      },
      "serial": 17,
      "x": 3.0139,
      "y": 28.6064,
      "z": 30.8744,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        16,
        24,
        18,
      ],
      "elem": "C",
      "index": 17,
      "properties": {
        "charge": 0.2136,
        "partialCharge": 0.2136,
      },
      "serial": 18,
      "x": 2.2228,
      "y": 27.7337,
      "z": 31.8944,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        17,
        19,
        26,
      ],
      "elem": "C",
      "index": 18,
      "properties": {
        "charge": 0.1834,
        "partialCharge": 0.1834,
      },
      "serial": 19,
      "x": 3.0305,
      "y": 26.5789,
      "z": 32.5223,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        14,
        18,
      ],
      "elem": "O",
      "index": 19,
      "properties": {
        "charge": -0.3262,
        "partialCharge": -0.3262,
      },
      "serial": 20,
      "x": 3.6985,
      "y": 25.8368,
      "z": 31.4989,
    },
    {
      "atom": "N.am",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        15,
        21,
      ],
      "elem": "N",
      "index": 20,
      "properties": {
        "charge": -0.1992,
        "partialCharge": -0.1992,
      },
      "serial": 21,
      "x": 4.9524,
      "y": 28.5919,
      "z": 29.2913,
    },
    {
      "atom": "C.2",
      "bondOrder": [
        1,
        2,
        1,
      ],
      "bonds": [
        20,
        22,
        23,
      ],
      "elem": "C",
      "index": 21,
      "properties": {
        "charge": 0.242,
        "partialCharge": 0.242,
      },
      "serial": 22,
      "x": 4.7764,
      "y": 29.2876,
      "z": 28.1519,
    },
    {
      "atom": "O.2",
      "bondOrder": [
        2,
      ],
      "bonds": [
        21,
      ],
      "elem": "O",
      "index": 22,
      "properties": {
        "charge": -0.2733,
        "partialCharge": -0.2733,
      },
      "serial": 23,
      "x": 3.8923,
      "y": 29.0778,
      "z": 27.4442,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        21,
      ],
      "elem": "C",
      "index": 23,
      "properties": {
        "charge": 0.0964,
        "partialCharge": 0.0964,
      },
      "serial": 24,
      "x": 5.7995,
      "y": 30.3429,
      "z": 27.9091,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        17,
      ],
      "elem": "O",
      "index": 24,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 25,
      "x": 1.7768,
      "y": 28.5816,
      "z": 33.0056,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        16,
      ],
      "elem": "O",
      "index": 25,
      "properties": {
        "charge": -0.2183,
        "partialCharge": -0.2183,
      },
      "serial": 26,
      "x": 2.0576,
      "y": 29.3423,
      "z": 30.0865,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        18,
        27,
      ],
      "elem": "C",
      "index": 26,
      "properties": {
        "charge": 0.2142,
        "partialCharge": 0.2142,
      },
      "serial": 27,
      "x": 2.1195,
      "y": 25.5825,
      "z": 33.2269,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        26,
      ],
      "elem": "O",
      "index": 27,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 28,
      "x": 0.7513,
      "y": 25.6243,
      "z": 32.7198,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        0,
        32,
      ],
      "elem": "O",
      "index": 28,
      "properties": {
        "charge": -0.3264,
        "partialCharge": -0.3264,
      },
      "serial": 29,
      "x": 6.2641,
      "y": 24.2063,
      "z": 24.7618,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        40,
        30,
        34,
      ],
      "elem": "C",
      "index": 29,
      "properties": {
        "charge": 0.2917,
        "partialCharge": 0.2917,
      },
      "serial": 30,
      "x": 8.6273,
      "y": 24.3604,
      "z": 21.3671,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        29,
        35,
        31,
      ],
      "elem": "C",
      "index": 30,
      "properties": {
        "charge": 0.187,
        "partialCharge": 0.187,
      },
      "serial": 31,
      "x": 8.9008,
      "y": 25.5399,
      "z": 22.3343,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        30,
        39,
        32,
      ],
      "elem": "C",
      "index": 31,
      "properties": {
        "charge": 0.2063,
        "partialCharge": 0.2063,
      },
      "serial": 32,
      "x": 8.2417,
      "y": 25.2652,
      "z": 23.67,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        31,
        28,
        33,
      ],
      "elem": "C",
      "index": 32,
      "properties": {
        "charge": 0.1795,
        "partialCharge": 0.1795,
      },
      "serial": 33,
      "x": 6.7494,
      "y": 24.8542,
      "z": 23.415,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        32,
        34,
        41,
      ],
      "elem": "C",
      "index": 33,
      "properties": {
        "charge": 0.1803,
        "partialCharge": 0.1803,
      },
      "serial": 34,
      "x": 6.4797,
      "y": 23.8858,
      "z": 22.2544,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        29,
        33,
      ],
      "elem": "O",
      "index": 34,
      "properties": {
        "charge": -0.3243,
        "partialCharge": -0.3243,
      },
      "serial": 35,
      "x": 7.1783,
      "y": 24.4097,
      "z": 21.065,
    },
    {
      "atom": "N.am",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        30,
        36,
      ],
      "elem": "N",
      "index": 35,
      "properties": {
        "charge": -0.1991,
        "partialCharge": -0.1991,
      },
      "serial": 36,
      "x": 10.3585,
      "y": 25.7246,
      "z": 22.4745,
    },
    {
      "atom": "C.2",
      "bondOrder": [
        1,
        2,
        1,
      ],
      "bonds": [
        35,
        37,
        38,
      ],
      "elem": "C",
      "index": 36,
      "properties": {
        "charge": 0.242,
        "partialCharge": 0.242,
      },
      "serial": 37,
      "x": 10.9478,
      "y": 26.9262,
      "z": 22.6921,
    },
    {
      "atom": "O.2",
      "bondOrder": [
        2,
      ],
      "bonds": [
        36,
      ],
      "elem": "O",
      "index": 37,
      "properties": {
        "charge": -0.2733,
        "partialCharge": -0.2733,
      },
      "serial": 38,
      "x": 10.3285,
      "y": 27.8881,
      "z": 22.7405,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        36,
      ],
      "elem": "C",
      "index": 38,
      "properties": {
        "charge": 0.0964,
        "partialCharge": 0.0964,
      },
      "serial": 39,
      "x": 12.4231,
      "y": 26.8477,
      "z": 22.8571,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        31,
      ],
      "elem": "O",
      "index": 39,
      "properties": {
        "charge": -0.2184,
        "partialCharge": -0.2184,
      },
      "serial": 40,
      "x": 8.2163,
      "y": 26.4191,
      "z": 24.5232,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        29,
      ],
      "elem": "O",
      "index": 40,
      "properties": {
        "charge": -0.1879,
        "partialCharge": -0.1879,
      },
      "serial": 41,
      "x": 9.2627,
      "y": 24.7279,
      "z": 20.1476,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        33,
        42,
      ],
      "elem": "C",
      "index": 41,
      "properties": {
        "charge": 0.2141,
        "partialCharge": 0.2141,
      },
      "serial": 42,
      "x": 5.0185,
      "y": 23.7807,
      "z": 21.8917,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        41,
      ],
      "elem": "O",
      "index": 42,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 43,
      "x": 4.1265,
      "y": 24.6034,
      "z": 22.78,
    },
  ],
]
`;

exports[`Function MOL2 | Input: multiple.mol2 | options: keepH | Atoms should match the snapshot (keepH: false) 1`] = `
[
  [
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        28,
        1,
        5,
      ],
      "elem": "C",
      "index": 0,
      "properties": {
        "charge": 0.2586,
        "partialCharge": 0.2586,
      },
      "serial": 1,
      "x": 5.9063,
      "y": 24.9584,
      "z": 25.9808,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        0,
        6,
        2,
      ],
      "elem": "C",
      "index": 1,
      "properties": {
        "charge": 0.1841,
        "partialCharge": 0.1841,
      },
      "serial": 2,
      "x": 4.7041,
      "y": 24.1561,
      "z": 26.7625,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        1,
        10,
        3,
      ],
      "elem": "C",
      "index": 2,
      "properties": {
        "charge": 0.2062,
        "partialCharge": 0.2062,
      },
      "serial": 3,
      "x": 4.3137,
      "y": 24.9723,
      "z": 28.0348,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        2,
        13,
        4,
      ],
      "elem": "C",
      "index": 3,
      "properties": {
        "charge": 0.1795,
        "partialCharge": 0.1795,
      },
      "serial": 4,
      "x": 5.6843,
      "y": 25.1844,
      "z": 28.7746,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        3,
        5,
        11,
      ],
      "elem": "C",
      "index": 4,
      "properties": {
        "charge": 0.1803,
        "partialCharge": 0.1803,
      },
      "serial": 5,
      "x": 6.7356,
      "y": 26.0149,
      "z": 27.8889,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        0,
        4,
      ],
      "elem": "O",
      "index": 5,
      "properties": {
        "charge": -0.3263,
        "partialCharge": -0.3263,
      },
      "serial": 6,
      "x": 7.1258,
      "y": 25.0806,
      "z": 26.8068,
    },
    {
      "atom": "N.am",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        1,
        7,
      ],
      "elem": "N",
      "index": 6,
      "properties": {
        "charge": -0.1992,
        "partialCharge": -0.1992,
      },
      "serial": 7,
      "x": 3.5113,
      "y": 23.9905,
      "z": 25.9254,
    },
    {
      "atom": "C.2",
      "bondOrder": [
        1,
        2,
        1,
      ],
      "bonds": [
        6,
        8,
        9,
      ],
      "elem": "C",
      "index": 7,
      "properties": {
        "charge": 0.242,
        "partialCharge": 0.242,
      },
      "serial": 8,
      "x": 3.1108,
      "y": 22.8071,
      "z": 25.3967,
    },
    {
      "atom": "O.2",
      "bondOrder": [
        2,
      ],
      "bonds": [
        7,
      ],
      "elem": "O",
      "index": 8,
      "properties": {
        "charge": -0.2733,
        "partialCharge": -0.2733,
      },
      "serial": 9,
      "x": 3.7384,
      "y": 21.8496,
      "z": 25.5073,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        7,
      ],
      "elem": "C",
      "index": 9,
      "properties": {
        "charge": 0.0964,
        "partialCharge": 0.0964,
      },
      "serial": 10,
      "x": 1.8037,
      "y": 22.8588,
      "z": 24.6767,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        2,
      ],
      "elem": "O",
      "index": 10,
      "properties": {
        "charge": -0.2184,
        "partialCharge": -0.2184,
      },
      "serial": 11,
      "x": 3.3693,
      "y": 24.2052,
      "z": 28.8011,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        4,
        12,
      ],
      "elem": "C",
      "index": 11,
      "properties": {
        "charge": 0.2141,
        "partialCharge": 0.2141,
      },
      "serial": 12,
      "x": 7.9504,
      "y": 26.3491,
      "z": 28.7097,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        11,
      ],
      "elem": "O",
      "index": 12,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 13,
      "x": 8.2497,
      "y": 25.1328,
      "z": 29.5559,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        3,
        14,
      ],
      "elem": "O",
      "index": 13,
      "properties": {
        "charge": -0.3264,
        "partialCharge": -0.3264,
      },
      "serial": 14,
      "x": 5.5738,
      "y": 25.9109,
      "z": 30.0347,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        13,
        15,
        19,
      ],
      "elem": "C",
      "index": 14,
      "properties": {
        "charge": 0.2586,
        "partialCharge": 0.2586,
      },
      "serial": 15,
      "x": 4.5765,
      "y": 26.6958,
      "z": 30.7444,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        14,
        20,
        16,
      ],
      "elem": "C",
      "index": 15,
      "properties": {
        "charge": 0.1842,
        "partialCharge": 0.1842,
      },
      "serial": 16,
      "x": 3.837,
      "y": 27.7677,
      "z": 29.9491,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        15,
        25,
        17,
      ],
      "elem": "C",
      "index": 16,
      "properties": {
        "charge": 0.2094,
        "partialCharge": 0.2094,
      },
      "serial": 17,
      "x": 3.0139,
      "y": 28.6064,
      "z": 30.8744,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        16,
        24,
        18,
      ],
      "elem": "C",
      "index": 17,
      "properties": {
        "charge": 0.2136,
        "partialCharge": 0.2136,
      },
      "serial": 18,
      "x": 2.2228,
      "y": 27.7337,
      "z": 31.8944,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        17,
        19,
        26,
      ],
      "elem": "C",
      "index": 18,
      "properties": {
        "charge": 0.1834,
        "partialCharge": 0.1834,
      },
      "serial": 19,
      "x": 3.0305,
      "y": 26.5789,
      "z": 32.5223,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        14,
        18,
      ],
      "elem": "O",
      "index": 19,
      "properties": {
        "charge": -0.3262,
        "partialCharge": -0.3262,
      },
      "serial": 20,
      "x": 3.6985,
      "y": 25.8368,
      "z": 31.4989,
    },
    {
      "atom": "N.am",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        15,
        21,
      ],
      "elem": "N",
      "index": 20,
      "properties": {
        "charge": -0.1992,
        "partialCharge": -0.1992,
      },
      "serial": 21,
      "x": 4.9524,
      "y": 28.5919,
      "z": 29.2913,
    },
    {
      "atom": "C.2",
      "bondOrder": [
        1,
        2,
        1,
      ],
      "bonds": [
        20,
        22,
        23,
      ],
      "elem": "C",
      "index": 21,
      "properties": {
        "charge": 0.242,
        "partialCharge": 0.242,
      },
      "serial": 22,
      "x": 4.7764,
      "y": 29.2876,
      "z": 28.1519,
    },
    {
      "atom": "O.2",
      "bondOrder": [
        2,
      ],
      "bonds": [
        21,
      ],
      "elem": "O",
      "index": 22,
      "properties": {
        "charge": -0.2733,
        "partialCharge": -0.2733,
      },
      "serial": 23,
      "x": 3.8923,
      "y": 29.0778,
      "z": 27.4442,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        21,
      ],
      "elem": "C",
      "index": 23,
      "properties": {
        "charge": 0.0964,
        "partialCharge": 0.0964,
      },
      "serial": 24,
      "x": 5.7995,
      "y": 30.3429,
      "z": 27.9091,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        17,
      ],
      "elem": "O",
      "index": 24,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 25,
      "x": 1.7768,
      "y": 28.5816,
      "z": 33.0056,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        16,
      ],
      "elem": "O",
      "index": 25,
      "properties": {
        "charge": -0.2183,
        "partialCharge": -0.2183,
      },
      "serial": 26,
      "x": 2.0576,
      "y": 29.3423,
      "z": 30.0865,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        18,
        27,
      ],
      "elem": "C",
      "index": 26,
      "properties": {
        "charge": 0.2142,
        "partialCharge": 0.2142,
      },
      "serial": 27,
      "x": 2.1195,
      "y": 25.5825,
      "z": 33.2269,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        26,
      ],
      "elem": "O",
      "index": 27,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 28,
      "x": 0.7513,
      "y": 25.6243,
      "z": 32.7198,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        0,
        32,
      ],
      "elem": "O",
      "index": 28,
      "properties": {
        "charge": -0.3264,
        "partialCharge": -0.3264,
      },
      "serial": 29,
      "x": 6.2641,
      "y": 24.2063,
      "z": 24.7618,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        40,
        30,
        34,
      ],
      "elem": "C",
      "index": 29,
      "properties": {
        "charge": 0.2917,
        "partialCharge": 0.2917,
      },
      "serial": 30,
      "x": 8.6273,
      "y": 24.3604,
      "z": 21.3671,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        29,
        35,
        31,
      ],
      "elem": "C",
      "index": 30,
      "properties": {
        "charge": 0.187,
        "partialCharge": 0.187,
      },
      "serial": 31,
      "x": 8.9008,
      "y": 25.5399,
      "z": 22.3343,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        30,
        39,
        32,
      ],
      "elem": "C",
      "index": 31,
      "properties": {
        "charge": 0.2063,
        "partialCharge": 0.2063,
      },
      "serial": 32,
      "x": 8.2417,
      "y": 25.2652,
      "z": 23.67,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        31,
        28,
        33,
      ],
      "elem": "C",
      "index": 32,
      "properties": {
        "charge": 0.1795,
        "partialCharge": 0.1795,
      },
      "serial": 33,
      "x": 6.7494,
      "y": 24.8542,
      "z": 23.415,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
        1,
      ],
      "bonds": [
        32,
        34,
        41,
      ],
      "elem": "C",
      "index": 33,
      "properties": {
        "charge": 0.1803,
        "partialCharge": 0.1803,
      },
      "serial": 34,
      "x": 6.4797,
      "y": 23.8858,
      "z": 22.2544,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        29,
        33,
      ],
      "elem": "O",
      "index": 34,
      "properties": {
        "charge": -0.3243,
        "partialCharge": -0.3243,
      },
      "serial": 35,
      "x": 7.1783,
      "y": 24.4097,
      "z": 21.065,
    },
    {
      "atom": "N.am",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        30,
        36,
      ],
      "elem": "N",
      "index": 35,
      "properties": {
        "charge": -0.1991,
        "partialCharge": -0.1991,
      },
      "serial": 36,
      "x": 10.3585,
      "y": 25.7246,
      "z": 22.4745,
    },
    {
      "atom": "C.2",
      "bondOrder": [
        1,
        2,
        1,
      ],
      "bonds": [
        35,
        37,
        38,
      ],
      "elem": "C",
      "index": 36,
      "properties": {
        "charge": 0.242,
        "partialCharge": 0.242,
      },
      "serial": 37,
      "x": 10.9478,
      "y": 26.9262,
      "z": 22.6921,
    },
    {
      "atom": "O.2",
      "bondOrder": [
        2,
      ],
      "bonds": [
        36,
      ],
      "elem": "O",
      "index": 37,
      "properties": {
        "charge": -0.2733,
        "partialCharge": -0.2733,
      },
      "serial": 38,
      "x": 10.3285,
      "y": 27.8881,
      "z": 22.7405,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        36,
      ],
      "elem": "C",
      "index": 38,
      "properties": {
        "charge": 0.0964,
        "partialCharge": 0.0964,
      },
      "serial": 39,
      "x": 12.4231,
      "y": 26.8477,
      "z": 22.8571,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        31,
      ],
      "elem": "O",
      "index": 39,
      "properties": {
        "charge": -0.2184,
        "partialCharge": -0.2184,
      },
      "serial": 40,
      "x": 8.2163,
      "y": 26.4191,
      "z": 24.5232,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        29,
      ],
      "elem": "O",
      "index": 40,
      "properties": {
        "charge": -0.1879,
        "partialCharge": -0.1879,
      },
      "serial": 41,
      "x": 9.2627,
      "y": 24.7279,
      "z": 20.1476,
    },
    {
      "atom": "C.3",
      "bondOrder": [
        1,
        1,
      ],
      "bonds": [
        33,
        42,
      ],
      "elem": "C",
      "index": 41,
      "properties": {
        "charge": 0.2141,
        "partialCharge": 0.2141,
      },
      "serial": 42,
      "x": 5.0185,
      "y": 23.7807,
      "z": 21.8917,
    },
    {
      "atom": "O.3",
      "bondOrder": [
        1,
      ],
      "bonds": [
        41,
      ],
      "elem": "O",
      "index": 42,
      "properties": {
        "charge": -0.2176,
        "partialCharge": -0.2176,
      },
      "serial": 43,
      "x": 4.1265,
      "y": 24.6034,
      "z": 22.78,
    },
  ],
]
`;
